### PR TITLE
Introduce golangci-lint and fix warnings

### PR DESCRIPTION
**Pull request by JAORMX**
This adds several fixes recommended by the tool.

The manual additions were for:

* errorlint
* errcheck
* stylecheck
* gosimple
* statischeck
* deadcode
* noctx
* unconvert
* maligned

Note that some of the fixes were auto-applied by the tool.

The golangci-lint is now added to the `verify` Makefile target and will
run on CI.

Some checks were commented out and will be addressed in subsequent PRs

### CODE REVIEW

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 5e848d1192ca152e7897d91af7a7c470a02cd9ea

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 5e848d1192ca152e7897d91af7a7c470a02cd9ea

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 075d1bc825cb0fc12a5a60636ca8489a243b783f

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 7cb28312c5ee838818d07e94d0daa3214dc2c319

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 2ae7844f4d9a7ffd4fabdf6edae5162846a60f61

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 13c96bcce956e9b51d3d3f48cdd6c8c7cd25b7ab

File: tests/e2e/helpers.go [hunk 1]
```diff
@@ -129,10 +132,14 @@ func (c *mcTestCtx) cleanupTrackedPools() {
 	for _, p := range c.pools {
 		// Then find all nodes that are labeled with this pool and remove the label
 		// Search the nodes with this label
-		poolNodes := getNodesWithSelector(c.f, p.Spec.NodeSelector.MatchLabels)
+		poolNodes, err := getNodesWithSelector(c.f, p.Spec.NodeSelector.MatchLabels)
+		if err != nil {
+			E2EErrorf(c.t, "Could get nodes with selector %s", p.Spec.NodeSelector.MatchLabels)
```
Comment (mrogers950) [inline tests/e2e/helpers.go:137]: s/Could/Could not

File: tests/e2e/helpers.go [hunk 2]
```diff
@@ -1301,7 +1348,11 @@ func createMachineConfigPoolSubset(t *testing.T, f *framework.Framework, oldPool
 	}
 
 	// list the nodes matching the node selector
-	poolNodes := getNodesWithSelector(f, oldPool.Spec.NodeSelector.MatchLabels)
+	poolNodes, err := getNodesWithSelector(f, oldPool.Spec.NodeSelector.MatchLabels)
+	if err != nil {
+		E2EErrorf(t, "Could get nodes with selector %s", oldPool.Spec.NodeSelector.MatchLabels)
```
Comment (mrogers950) [inline tests/e2e/helpers.go:1353]: s/Could/Could not

File: tests/e2e/helpers.go [hunk 3]
```diff
@@ -678,14 +694,31 @@ func suiteErrorMessageMatchesRegex(t *testing.T, f *framework.Framework, namespa
...
+func getNodesWithSelector(f *framework.Framework, labelselector map[string]string) ([]corev1.Node, error) {
 	var nodes corev1.NodeList
 	lo := &client.ListOptions{
 		LabelSelector: labels.SelectorFromSet(labelselector),
 	}
-	f.Client.List(goctx.TODO(), &nodes, lo)
-	return nodes.Items
+
+	var lastErr error
+	// retry and ignore errors until timeout
```
Comment (mrogers950) [inline tests/e2e/helpers.go:705]: Was this a linter recommendation? We might want to add to the function comment that it will now wait on results.

**Comment by JAORMX** [comment]
I'll introduce this later. Seems quite risky to do it now.

**Pull request closed without merging**